### PR TITLE
Added Hanning kernel, prepare for KCF tracking

### PR DESCRIPTION
**Pull request by ducha-aiki**
#### Changes
Analogue of https://numpy.org/doc/stable/reference/generated/numpy.hanning.html

Needed for https://github.com/kornia/kornia/issues/1381

#### Type of change

- [] 📚  Documentation Update
- [x] 🧪 Tests Cases
- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [x] 🔬 New feature (non-breaking change which adds functionality)
- [ ] 🚨 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📝 This change requires a documentation update

#### Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] Did you update CHANGELOG in case of a major change?

### CODE REVIEW

File: kornia/filters/kernels.py [hunk 1]
```diff
@@ -3,6 +3,8 @@
 
 import torch
 
+from kornia.constants import pi
```
Comment (edgarriba) [inline kornia/filters/kernels.py:6]:
```suggestion
import math
```

File: kornia/filters/kernels.py [hunk 2]
```diff
@@ -630,3 +632,53 @@ def get_hysteresis_kernel(device=torch.device('cpu'), dtype=torch.float) -> torc
         dtype=dtype,
     )
     return kernel.unsqueeze(1)
+
+
+def get_hanning_kernel1d(kernel_size: int, device=torch.device('cpu'), dtype=torch.float) -> torch.Tensor:
+    r"""Returns Hanning (also known as Hann) kernel, used in signal processing and KCF tracker
```
Comment (edgarriba) [inline kornia/filters/kernels.py:638]:
```suggestion
    r"""Returns Hanning (also known as Hann) kernel, used in signal processing and KCF tracker.

```

File: kornia/filters/kernels.py [hunk 3]
```diff
@@ -630,3 +632,53 @@ def get_hysteresis_kernel(device=torch.device('cpu'), dtype=torch.float) -> torc
...
+    .. math::  w(n) = 0.5 - 0.5cos\\left(\\frac{2\\pi{n}}{M-1}\\right)
+               \\qquad 0 \\leq n \\leq M-1
```
Comment (edgarriba) [inline kornia/filters/kernels.py:640]:
```suggestion
               \\qquad 0 \\leq n \\leq M-1

```

File: kornia/filters/kernels.py [hunk 4]
```diff
@@ -630,3 +632,53 @@ def get_hysteresis_kernel(device=torch.device('cpu'), dtype=torch.float) -> torc
...
+    See further in numpy docs https://numpy.org/doc/stable/reference/generated/numpy.hanning.html
+    Args:
```
Comment (edgarriba) [inline kornia/filters/kernels.py:642]:
```suggestion

    Args:
```

File: kornia/filters/kernels.py [hunk 5]
```diff
@@ -630,3 +632,53 @@ def get_hysteresis_kernel(device=torch.device('cpu'), dtype=torch.float) -> torc
...
+        kernel_size: It should be positive.
```
Comment (edgarriba) [inline kornia/filters/kernels.py:643]:
```suggestion
        kernel_size: The size the of the kernel. It should be positive.
```

File: kornia/filters/kernels.py [hunk 6]
```diff
@@ -630,3 +632,53 @@ def get_hysteresis_kernel(device=torch.device('cpu'), dtype=torch.float) -> torc
...
+
+    Examples:
+        >>> get_hanning_kernel1d(4)
+        tensor([ 0., 0.75, 0.75, 0.])
+    """
+    if not isinstance(kernel_size, int) or kernel_size <= 2:
+        raise TypeError(f"ksize must be an positive integer > 2. Got {kernel_size}")
+
+    x: torch.Tensor = torch.arange(kernel_size, device=device, dtype=dtype)
+    x = 0.5 - 0.5 * torch.cos(2.0 * pi * x / float(kernel_size - 1))
```
Comment (edgarriba) [inline kornia/filters/kernels.py:660]:
```suggestion
    x = 0.5 - 0.5 * torch.cos(2.0 * math.pi * x / float(kernel_size - 1))
```

File: kornia/filters/kernels.py [hunk 7]
```diff
@@ -630,3 +632,53 @@ def get_hysteresis_kernel(device=torch.device('cpu'), dtype=torch.float) -> torc
...
+    return x
+
+
+def get_hanning_kernel2d(kernel_size: Tuple[int, int],
+                         device=torch.device('cpu'),
+                         dtype=torch.float) -> torch.Tensor:
+    r"""Returns 2d Hanning kernel, used in signal processing and KCF tracker
```
Comment (edgarriba) [inline kornia/filters/kernels.py:667]:
```suggestion
    r"""Returns 2d Hanning kernel, used in signal processing and KCF tracker.

```

File: kornia/filters/kernels.py [hunk 8]
```diff
@@ -630,3 +632,53 @@ def get_hysteresis_kernel(device=torch.device('cpu'), dtype=torch.float) -> torc
...
+    Args:
+        kernel_size: It should be positive.
```
Comment (edgarriba) [inline kornia/filters/kernels.py:669]:
```suggestion
        kernel_size: The size of the kernel for the filter. It should be positive.
```

**Review by edgarriba — CHANGES_REQUESTED** [review]
Add also to docs

File: kornia/filters/kernels.py (hunk 1 above)
Comment (ducha-aiki) [reply kornia/filters/kernels.py:6]: Why do we have pi in kornia then?

Comment (edgarriba) [reply kornia/filters/kernels.py:6]: in initial pytorch versions floats were not really promoted with tensors

**Review by edgarriba — APPROVED** [review]

**Comment by ducha-aiki** [comment]
@edgarriba why the test are failing because of "FAILED test/augmentation/test_container.py::TestAugmentationSequential::test_random_crops[cpu-float32]"?

**Comment by edgarriba** [comment]
We should merge after this: https://github.com/kornia/kornia/pull/1521

**Pull request merged**